django_select2/forms.py

```diff
@@ -408,8 +409,7 @@ def render_options(self, *args):
             choices = {(obj.pk, self.label_from_instance(obj))
                        for obj in self.choices.queryset.filter(pk__in=selected_choices)}
         else:
-            choices = {(k, v) for k, v in choices if k in selected_choices}
-        selected_choices = {force_text(v) for v in selected_choices}
+            choices = {(k, v) for k, v in choices if force_text(k) in selected_choices}
```

Comment:
If I remember right it should be the same in this case. nevertheless here the code is going in line with what is django using in their code in Select widget.